tests/test_client.py

```diff
@@ -132,71 +185,92 @@ def test_from_token():
     assert quetz_client.session.headers.get("X-API-Key") == token
 
 
-def test_set_channel(
-    quetz_client: QuetzClient,
+def test_mock_set_channel(
+    mock_client: QuetzClient,
     requests_mock,
-    test_url: str,
+    mock_server: str,
 ):
     channel = "a"
 
-    url = f"{test_url}/api/channels"
+    url = f"{mock_server}/api/channels"
     requests_mock.post(url, json=None)
 
-    quetz_client.set_channel(channel)
+    mock_client.set_channel(channel)
 
     last_request = requests_mock.request_history[0]
     assert last_request.method == "POST"
     assert last_request.json()["name"] == channel
 
 
-def test_delete_channel(
-    quetz_client: QuetzClient,
+def test_mock_delete_channel(
+    mock_client: QuetzClient,
     requests_mock,
-    test_url: str,
+    mock_server: str,
 ):
     channel = "a"
 
-    url = f"{test_url}/api/channels/{channel}"
+    url = f"{mock_server}/api/channels/{channel}"
     requests_mock.delete(url, json=None)
 
-    quetz_client.delete_channel(channel)
+    mock_client.delete_channel(channel)
 
     last_request = requests_mock.request_history[0]
     assert last_request.method == "DELETE"
 
 
-def test_yield_packages(quetz_client: QuetzClient, expected_packages):
+def test_mock_yield_packages(mock_client: QuetzClient, expected_packages):
     channel = "channel1"
     package_set = {
-        (p.name, p.url, p.current_version) for p in quetz_client.yield_packages(channel)
+        (p.name, p.url, p.current_version) for p in mock_client.yield_packages(channel)
     }
     assert {
         (ep["name"], ep["url"], ep["current_version"])
         for ep in expected_packages["result"]
     } == package_set
 
 
-def test_post_file_to_channel(
-    quetz_client: QuetzClient,
+def test_mock_post_file_to_channel(
+    mock_client: QuetzClient,
     requests_mock,
-    test_url: str,
+    mock_server: str,
 ):
     channel = "a"
 
-    url_matcher = re.compile(
-        f"{test_url}/api/channels/{channel}/upload/\\w*\\?force=False&sha256=\\w*"
+    url = f"{mock_server}/api/channels/{channel}/files/"
+    requests_mock.post(url, json=None)
+
+    requests_mock.register_uri(
+        "GET",
+        "https://conda.anaconda.org/conda-forge/linux-64/xtensor-0.16.1-0.tar.bz2",
+        real_http=True,
     )
-    requests_mock.register_uri("POST", url_matcher, json=None)
 
+    with temporary_package_file() as file:
+        mock_client.post_file_to_channel(channel, file)
+
+    # the last request here might be the download of the test package file
+    # thus we need to access all the requests
+    assert len(requests_mock.request_history) <= 2
+    assert any(r.method == "POST" for r in requests_mock.request_history)
+
+
+def test_live_post_file_to_channel(
+    live_client: QuetzClient,
+    live_post_channel_a,
+    requests_mock,
+):
     requests_mock.register_uri(
         "GET",
         "https://conda.anaconda.org/conda-forge/linux-64/xtensor-0.16.1-0.tar.bz2",
         real_http=True,
     )
```

Comment:
why is this necessary?

Reply:
I'm not sure. I would think that by using `real_http=True` in `live_client` in `conftest.py`, this would be enough, but it's not for some reason. I added a comment.